### PR TITLE
topology2: ssp: Add blob version

### PR DESCRIPTION
**Pull request by ranj063**
Add the SSP blob version and set it based on the platform.

This is a replacement for https://github.com/thesofproject/sof/pull/7060

Fixes: https://github.com/thesofproject/sof/issues/6837

### CODE REVIEW

**Review by aiChaoSONG — APPROVED** [review]
Changes look good and simpler than #7060.

In the SSP context, the word `version` maybe too general, I think `blob_version` maybe better, but no big deal.

File: tools/topology/topology2/platform/intel/tgl.conf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# TGL/ADL-specific variable definitions
+Define {
+	SSP_BLOB_VERSION		0x100
```
[the remark this answers is not in the document]
Comment (ranj063) [reply tools/topology/topology2/platform/intel/tgl.conf:3]: @plbossart I think blob version 1.00 for TGL/ADL is what makes it possible for the alsatplg compiler to ignore the aux parameters that include the link clock_source

File: tools/topology/topology2/common_definitions.conf [hunk 2]
```diff
@@ -58,4 +58,7 @@ Define {
 	SPI_INPUT_CLASS				26 # SPI Input (DSP <-)
 
 	DEEPBUFFER_FW_DMA_MS			100 # 100 ms copier dma size
+
+	SSP_BLOB_VERSION_1_0			0x100
+	SSP_BLOB_VERSION_1_5			0x105
```
Comment (lgirdwood) [inline tools/topology/topology2/common_definitions.conf:63]: 0x150 ?

**Review by kv2019i — APPROVED** [review]
Not entirely sure how well this aligns with https://github.com/thesofproject/sof/pull/7370 , but put these two together seems to be a clean solution. Approving at least this one.

File: tools/topology/topology2/common_definitions.conf (hunk 2 above)
Comment (plbossart) [reply tools/topology/topology2/common_definitions.conf:63]: ack, this is 105 - as strange as it might look

File: tools/topology/topology2/platform/intel/tgl.conf (hunk 1 above)
[the remark this answers is not in the document]
Comment (plbossart) [reply tools/topology/topology2/platform/intel/tgl.conf:3]: @kv2019i I don't think that's true any longer with the change suggested by @ranj063 to make zephyr deal with aux in all cases, and do nothing for TGL.

[the remark this answers is not in the document]
Comment (plbossart) [reply tools/topology/topology2/platform/intel/tgl.conf:3]: @kv2019i see https://github.com/zephyrproject-rtos/zephyr/pull/56497

[the remark this answers is not in the document]
Comment (plbossart) [reply tools/topology/topology2/platform/intel/tgl.conf:3]: I think the SSP blob 1.05 was added before TGL. @mmaka1 @mwasko can you please confirm?

**Comment by ranj063** [comment]
@plbossart @kv2019i @lgirdwood I got to the root of the problem:
1. There's a strict restriction wrt the alsatplg compiler which requires the "Object.Base.link_config.1" to be defined right within the Object.Base.hw_config.N{} node. This is wrong and should be fixed in the compiler so that the Object definition in hw_config_cardinal_clock.conf will work as expected.
2. blob version 0x100 is required for TGL with the above change so that the clock_source setting will be ignored.
3. Alternatively, if we set _c105 for TGL, we need the zephyr patch https://github.com/zephyrproject-rtos/zephyr/pull/56497

In conclusion, I think this is what we should do. Merge this PR so that we can unblock the alsabat tests in CI. Then work on fixing the alsatplg to allow link_config to come from the class definition instead of the object instance. Finally after the zephyr patch is merged, set the blob type to 0x105 for all platforms.

Let me know what you think.

File: tools/topology/topology2/platform/intel/bt-ssp-config.conf [hunk 3]
```diff
@@ -50,6 +56,9 @@ Object.Dai.SSP [
 			tdm_slots	2
 			tx_slots	3
 			rx_slots	0
+			Object.Base.link_config.1 {
+				clock_source	0
```
Comment (kv2019i) [inline tools/topology/topology2/platform/intel/bt-ssp-config.conf:60]: Should this be the cardinal clock (i.e. "1"), so we can use integer divider for 1.536Mhz?

File: tools/topology/topology2/platform/intel/bt-ssp-config-lbm.conf [hunk 4]
```diff
@@ -51,6 +57,9 @@ Object.Dai.SSP [
 			tdm_slots	2
 			tx_slots	3
 			rx_slots	3
+			Object.Base.link_config.1 {
+				clock_source	0
```
Comment (kv2019i) [inline tools/topology/topology2/platform/intel/bt-ssp-config-lbm.conf:61]: See comment for non-loopback version.

**Review by kv2019i — COMMENTED** [review]
One comment inline about BT loopback topologies. Otherwise this looks like a clean approach.

File: tools/topology/topology2/platform/intel/bt-ssp-config.conf (hunk 3 above)
Comment (ranj063) [reply tools/topology/topology2/platform/intel/bt-ssp-config.conf:60]: no should 0 for xtal as xtal is 38.4 and 38.4M/1.536M = 25

**Comment by fredoh9** [comment]
I check this PR with MTL nocodec device, it generates solid waveform, alsabat test consistently passes!

File: tools/topology/topology2/platform/intel/bt-ssp-config.conf (hunk 3 above)
Comment (ranj063) [reply tools/topology/topology2/platform/intel/bt-ssp-config.conf:60]: @kv2019i I will let you answer this question and follow up with another PR if we need to change the clock source for BT. For now, it actually makes no difference since we dont build the topology for MTL
